### PR TITLE
Update jetty-maven-plugin to match winstone

### PR DESCRIPTION
**Pull request by daniel-beck**
Otherwise debugging inside Jetty is a mess in IntelliJ 😦 

### Proposed changelog entries

* Internal: Update maven-jetty-plugin used when debugging jenkins-war

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]

File: war/pom.xml [hunk 1]
```diff
@@ -104,6 +104,7 @@ THE SOFTWARE.
       <groupId>org.jenkins-ci</groupId>
       <artifactId>winstone</artifactId>
       <version>5.14</version>
+      <!-- Make sure to keep maven-jetty-plugin elsewhere in this file in sync with the Jetty release in Winstone -->
```
Comment (daniel-beck) [inline war/pom.xml:107]:
```suggestion
      <!-- Make sure to keep jetty-maven-plugin elsewhere in this file in sync with the Jetty release in Winstone -->
```

🤦

**Review by res0nance — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

File: war/pom.xml [hunk 2]
```diff
@@ -450,7 +451,7 @@ THE SOFTWARE.
       <plugin>
         <groupId>org.eclipse.jetty</groupId>
         <artifactId>jetty-maven-plugin</artifactId>
-        <version>9.4.22.v20191022</version>
+        <version>9.4.37.v20210219</version>
```
Comment (olamy) [inline war/pom.xml:454]:
```suggestion
        <version>9.4.38.v20210224</version>
```

**Review by oleg-nenashev — APPROVED** [review]
+1 after the update to 9.4.38

File: war/pom.xml (hunk 2 above)
Comment (daniel-beck) [reply war/pom.xml:454]: Eh, this PR will conflict with yours anyway as we touch adjacent lines.

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

File: war/pom.xml [hunk 3]
```diff
@@ -103,6 +103,7 @@ THE SOFTWARE.
       -->
       <groupId>org.jenkins-ci</groupId>
       <artifactId>winstone</artifactId>
+      <!-- TODO: Make sure to keep jetty-maven-plugin elsewhere in this file in sync with the Jetty release in Winstone -->
```
Comment (daniel-beck) [inline war/pom.xml:106]:
```suggestion
      <!-- Make sure to keep jetty-maven-plugin elsewhere in this file in sync with the Jetty release in Winstone -->
```

There's nothing _to do_ here in terms of open tasks, it's just a reminder.

**Pull request merged**